test/test_schema.py

```diff
@@ -290,3 +291,29 @@ def fn(q: InputC) -> None:
     }
     assert input_schema == expected_input_schema
     assert is_validate_dict(input_schema)
+
+
+def test_schema_to_base_model():
+    def fn(
+        a: int,
+        b: Union[float, int] = 4,
```

Comment:
New typing method e.g. `b: float | int = 4` is not supported and yields error. Underlying code expects union type of `GenericAlias`, but new typing is an direct instance of `UnionType`. Please see following example:
```
>>> from typing import Union
>>> new_typing = int | str
>>> old_typing = Union[int, str]
>>> new_typing == old_typing
True
>>> type(new_typing)
<class 'types.UnionType'>
>>> type(old_typing)
<class 'typing._UnionGenericAlias'>
```

Reply:
Good catch, this is now fixes. Haven't used this feature of 3.12 yet so forgot it existed.